conanfile.py

```diff
@@ -123,24 +124,44 @@ def system_requirements(self):
         # TODO: Remove this: requirements and optional requirements are
         # installed automatically by add_basilisk_to_sys_path(). Only build
         # system requirements need to be installed here.
-        reqFile = open('requirements.txt', 'r')
-        required = reqFile.read().replace("`", "").split('\n')
-        reqFile.close()
-        pkgList = [x.lower() for x in required]
-
-        reqFile = open('requirements_dev.txt', 'r')
-        required = reqFile.read().replace("`", "").split('\n')
-        reqFile.close()
-        pkgList += [x.lower() for x in required]
-
-        checkStr = "Required"
-        if self.options.get_safe("allOptPkg"):
-            optFile = open('requirements_doc.txt', 'r')
-            optionalPkgs = optFile.read().replace("`", "").split('\n')
-            optFile.close()
-            optionalPkgs = [x.lower() for x in optionalPkgs]
-            pkgList += optionalPkgs
-            checkStr += " and All Optional"
+        if self.options.get_safe("pyPkgCanary"):
+            reqFile = open('.github/workflows/requirements_canary.txt', 'r')
```

Comment:
I would keep the same file names `requirements.txt`.  This way you can simply prepend the path to this file and keep this path empty if we are not running in canary mode?